src/intersectables/intersectableList.cpp

```diff
@@ -23,11 +24,49 @@ HitRecord* IntersectableList::intersect(const Ray& ray) const {
     }
 
     // only delete the current hitRecord if there is more than one intersectable in the scene
-    if (container->size() > 1) {
+    if (container.size() > 1) {
       delete currentHitRecord;
     }
   }
 
   return hitRecord;
 }
-void IntersectableList::put(Intersectable* intersectable) { container->push_back(intersectable); }
+
+const BoundingBox& IntersectableList::getBoundingBox() const { return aabb; }
+
+BoundingBox IntersectableList::computeBoundingBox() {
+  float minT = std::numeric_limits<float>::min();
+  float maxT = std::numeric_limits<float>::max();
+
+  auto minX = maxT;
+  auto minY = maxT;
+  auto minZ = maxT;
+
+  auto maxX = minT;
+  auto maxY = minT;
+  auto maxZ = minT;
+
+  for (auto const& intersectable : container) {
+    auto boundingBox = intersectable->getBoundingBox();
+    auto bottomLeft = boundingBox.getBottomLeft();
+    auto topRight = boundingBox.getTopRight();
+
+    minX = std::min((double)bottomLeft.x, (double)minX);
+    minY = std::min((double)bottomLeft.y, (double)minY);
+    minZ = std::min((double)bottomLeft.z, (double)minZ);
+
+    maxX = std::max((double)topRight.x, (double)maxX);
+    maxY = std::max((double)topRight.y, (double)maxY);
+    maxZ = std::max((double)topRight.z, (double)maxZ);
+  }
+
+  return BoundingBox(Vector3f(minX, minY, minZ), Vector3f(maxX, maxY, maxZ));
+}
+
+void IntersectableList::put(Intersectable* intersectable) {
+  container.push_back(intersectable);
+
+  // update bounding box
+  auto newBoundingBox = computeBoundingBox();
```

Comment:
This changes `put` from an O(1) operation to an O(n). 

That's not necessary: Just add an `expand(Intersectable* i)` method to BoundingBox that enlarges it to also contain the given intersectable.

Reply:
this makes sense - thanks for spotting this :)